radio/src/gui/480x272/widgets/outputs.cpp

```diff
@@ -57,8 +53,18 @@ class OutputsWidget: public Widget
           uint16_t size = divRoundClosest(- RECT_WIDTH * chanVal, 200);
           lcdDrawSolidFilledRect(endpoint - size,  y + (curChan - firstChan) * row_height, size, row_height, MAINVIEW_GRAPHICS_COLOR);
         }
-        lcdDrawRect(x + RECT_OFFSET, y + (curChan - firstChan) * row_height, RECT_WIDTH, row_height+1);
         lcd->drawSolidVerticalLine(x + RECT_OFFSET + RECT_WIDTH / 2, y + (curChan - firstChan) * row_height, row_height, MAINVIEW_GRAPHICS_COLOR);
+        lcdDrawNumber(x + RECT_WIDTH - 10, y + (curChan - firstChan) * row_height + 1, chanVal, SMLSIZE | TEXT_COLOR | RIGHT, 0, NULL, "%");
+        if (g_model.limitData[curChan - 1].name[0] != 0)  {
+          strAppendSigned(chanString, curChan, 2);
+          lcdDrawText(x + 2, y + (curChan - firstChan) * row_height + 1, chanString, SMLSIZE | TEXT_COLOR | LEFT);
+          lcdDrawSizedText(x + 25, y + (curChan - firstChan) * row_height + 1, g_model.limitData[curChan - 1].name, sizeof(g_model.limitData[curChan - 1].name), SMLSIZE | TEXT_COLOR | LEFT | ZCHAR);
+      } 
+        else {
+          strAppend(chanString, "CH");
+          strAppendSigned(&chanString[2], curChan, 2);
```

Comment:
You are relying on first two chars of `chanString` to be `CH`, but you could already overwritten that on line 59!